clarity/src/vm/database/clarity_db.rs

```diff
@@ -903,13 +907,32 @@ impl<'a> ClarityDatabase<'a> {
             .ok_or_else(|| InterpreterError::Expect("Failed to get block data.".into()).into())
     }
 
-    pub fn get_block_time(&mut self, block_height: u32) -> Result<u64> {
+    pub fn get_burn_block_time(&mut self, block_height: u32) -> Result<u64> {
         let id_bhh = self.get_index_block_header_hash(block_height)?;
         self.headers_db
             .get_burn_block_time_for_block(&id_bhh)
             .ok_or_else(|| InterpreterError::Expect("Failed to get block data.".into()).into())
     }
 
+    pub fn get_block_time(&mut self, block_height: u32) -> Result<u64> {
+        let id_bhh = self.get_index_block_header_hash(block_height)?;
+        let burn_block_height = self
+            .get_burnchain_block_height(&id_bhh)
+            .ok_or_else(|| InterpreterError::Expect("Failed to get block data.".into()))?;
+        let epoch = self.get_stacks_epoch(burn_block_height).ok_or_else(|| {
+            InterpreterError::Expect(
+                format!("Failed to get epoch for block height {block_height}.)").into(),
+            )
+        })?;
+        if !epoch.epoch_id.has_block_timestamps() {
+            return self.get_burn_block_time(block_height);
```

Comment:
Can you pass `id_bhh` here so you don't have to look it up again?

Reply:
You mean add `id_bhh` as an optional parameter? We don't already have it at the other call sites.

Reply:
Yeah, an optional parameter will do.

Reply:
Added in 5b1191cf3927dabbc7516507915037f052425170.